block/submit.go

```diff
@@ -49,37 +49,64 @@ func (m *Manager) handleSubmissionTrigger(ctx context.Context) {
 		m.logger.Error("produce empty block", "error", err)
 	}
 
-	syncHeight, err := m.submitNextBatch()
+	if m.pendingBatch != nil {
+		m.logger.Info("pending batch exists", "startHeight", m.pendingBatch.batch.StartHeight, "endHeight", m.pendingBatch.batch.EndHeight)
+	} else {
+		nextBatch, err := m.createNextBatch()
+		if err != nil {
+			m.logger.Error("get next batch", "error", err)
+			return
+		}
+
+		resultSubmitToDA, err := m.submitNextBatchToDA(nextBatch)
+		if err != nil {
+			m.logger.Error("submit next batch", "error", err)
+			return
+		}
+
+		m.pendingBatch = &struct {
```

Comment:
I think would be cleaner to define the struct not in-place as it requires re-declearing the fields. or alternatively initiate it on the newManager vs on every submit attempt